python/tank/templatekey.py

```diff
@@ -256,6 +272,124 @@ def _as_string(self, value):
         return value if isinstance(value, basestring) else str(value)
 
 
+class TimestampKey(TemplateKey):
+    """
+    Key whose value is a time string formatted with strftime.
+    """
+
+    def __init__(
+        self,
+        name,
+        default=None,
+        format_spec="%Y-%m-%d-%H-%M-%S"
+    ):
+        """
+        Constructor
+        :param name: Name by which the key will be refered.
+        :param default: Default value for this field. Acceptable values are
+                            - None
+                            - a datetime.datetime object
```

Comment:
I do it once in unit tests. There's no use case for it, I agree. I'll take it out.